swift-sdk/Internal/InAppInternal.swift

```diff
@@ -5,59 +5,42 @@
 
 import Foundation
 
-/// Callbacks from the synchronizer
-protocol InAppSynchronizerDelegate : class {
-    func onInAppRemoved(messageId: String)
-    func onInAppMessagesAvailable(messages: [IterableInAppMessage])
-}
-
 ///
 protocol InAppSynchronizerProtocol {
-    // These variables are used for callbacks
-    var internalApi: IterableAPIInternal? {get set}
-    var inAppSyncDelegate: InAppSynchronizerDelegate? {get set}
-    
-    // These methods are called on new messages arrive etc.
-    func sync()
-    func remove(messageId: String)
+    // Fetch from server and sync
+    func sync() -> Future<[IterableInAppMessage]>
+}
+
+/// For callbacks when silent push notifications arrive
+protocol InAppNotifiable {
+    func onInAppSyncNeeded()
+    func onInAppRemoved(messageId: String)
 }
 
 extension IterableInAppTriggerType {
     static let defaultTriggerType = IterableInAppTriggerType.immediate // default is what is chosen by default
     static let undefinedTriggerType = IterableInAppTriggerType.never // undefined is what we select if payload has new trigger type
 }
 
-class InAppSilentPushSynchronizer : InAppSynchronizerProtocol {
-    weak var internalApi: IterableAPIInternal?
-    weak var inAppSyncDelegate: InAppSynchronizerDelegate?
-    
+class InAppSynchronizer : InAppSynchronizerProtocol {
```

Comment:
Looking at the new implementation, this class isn't doing any synchronization and just returns a server-side queue now. What do you think about renaming it to something like `InAppFetcher`?